### PR TITLE
Cta ga tracking

### PR DESCRIPTION
**Pull request by hngerebara**
- Add ga tracking to links within the scribe modal 
- Add ga track on click of the subscribe link

Screenshot 2019-10-04 at 3 19 27 AM
Screenshot 2019-10-04 at 3 01 18 AM

### CODE REVIEW

File: _includes/subscribe-modal.html [hunk 1]
```diff
@@ -19,23 +19,23 @@ <h3 class="modal-title no-anchor">Gruntwork Service Catalog</h3>
...
-                            <a href="/support/">Learn More</a>
+                            <a href="/support/" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-support">Learn More</a>
                         </p>
                     </li>
                 </ul>
             </div>
             <div class="modal-footer">
                 <div>
-                    <a class="btn btn-primary" href="/pricing/" role="button">Continue</a>
+                    <a class="btn btn-primary" href="/pricing/" role="button" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-pricing">Continue</a>
```
Comment (brikis98) [inline _includes/subscribe-modal.html]:
```suggestion
                    <a class="btn btn-primary" href="/pricing/" role="button" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="continue">Continue</a>
```

File: _includes/subscribe-modal.html [hunk 2]
```diff
@@ -19,23 +19,23 @@ <h3 class="modal-title no-anchor">Gruntwork Service Catalog</h3>
                     <li>
                         <h3 class="modal-title no-anchor">Gruntwork Training Library</h3>
                         <p>Learn Terraform, Docker, Packer, and other DevOps topics from a series of video courses. Learn More 
-                            <a href="/training/">Learn More</a>
+                            <a href="/training/" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-training">Learn More</a>
```
Comment (brikis98) [inline _includes/subscribe-modal.html]:
```suggestion
                            <a href="/training/" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="training-learn-more">Learn More</a>
```

File: _includes/subscribe-modal.html [hunk 3]
```diff
@@ -19,23 +19,23 @@ <h3 class="modal-title no-anchor">Gruntwork Service Catalog</h3>
...
                         </p>
                     </li>
                     <li>
                         <h3 class="modal-title no-anchor">Gruntwork Support</h3>
                         <p>Get help via email, chat, and phone/video from a team of DevOps experts.
-                            <a href="/support/">Learn More</a>
+                            <a href="/support/" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-support">Learn More</a>
```
Comment (brikis98) [inline _includes/subscribe-modal.html]:
```suggestion
                            <a href="/support/" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="support-learn-more">Learn More</a>
```

File: _includes/subscribe-modal.html [hunk 4]
```diff
@@ -19,23 +19,23 @@ <h3 class="modal-title no-anchor">Gruntwork Service Catalog</h3>
...
                 </ul>
             </div>
             <div class="modal-footer">
                 <div>
-                    <a class="btn btn-primary" href="/pricing/" role="button">Continue</a>
+                    <a class="btn btn-primary" href="/pricing/" role="button" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-pricing">Continue</a>
                 </div>
                 <div>
-                    <a class="btn btn-link" href="/contact/" role="button">Contact Sales</a>
+                    <a class="btn btn-link" href="/contact/" role="button" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="subscribe-modal-contact">Contact Sales</a>
```
Comment (brikis98) [inline _includes/subscribe-modal.html]:
```suggestion
                    <a class="btn btn-link" href="/contact/" role="button" ga-on="click" ga-event-category="subscribe-modal-{{ page.path | slugify}}" ga-event-action="contact-sales">Contact Sales</a>
```

File: assets/js/guides.js [hunk 5]
```diff
@@ -113,4 +113,12 @@ $(document).ready(function () {
   $('.post-detail img').on('click', function() {
     window.open(this.src, '_blank')
   })
-});
\ No newline at end of file
+
+  //Add ga tracking on clicking of the subscribe cta within the guides
+  $('.admonitionblock a').on('click', function() {
+    if($(this).text().toLowerCase().includes('subscriber')){
+      ga('send', 'event', location.pathname, 'cta-subscribe');
```
Comment (brikis98) [inline assets/js/guides.js]: Hm, this seems a bit too magical and brittle... 

I believe our goal should be:

1. For the "you must be a subscriber" CTA, we pop-up the modal. It looks like this code only sends tracking?
1. We want to do show this modal and capture these clicks in a way that isn't too brittle. 

Let's instead add [AsciiDoc roles](https://asciidoctor.org/docs/user-manual/#role) to those links in the deployment guides. Those roles translate to CSS class names that we can target more explicitly.

Comment (hngerebara) [reply assets/js/guides.js]: I am triggering the modal [here](https://github.com/gruntwork-io/gruntwork-io.github.io/blob/master/assets/js/main.js#L49) and yes the code sends only tracking. I've modified the code to retrieve the class name instead

**Comment by hngerebara** [comment]
@brikis98 I made some changes to this PR some days ago. Please, can you take a look? Thanks

**Comment by brikis98** [comment]
Sure, will look now. As a meta-comment, I really appreciate explicit nudges to re-review something, as it's not always obvious when a PR is ready for another round. Thx!

**Pull request merged**